src/rgw/rgw_coroutine.cc

```diff
@@ -94,6 +94,7 @@ void RGWCompletionManager::go_down()
   Mutex::Locker l(lock);
   for (auto cn : cns) {
     cn->unregister();
+    cn->put();
```

Comment:
these references in `cns` are already managed by `boost::intrusive_ptr`, so we shouldn't need to release them here manually - they'll go away automatically when the destructor runs. if we need to release them early, here in `go_down()`, you can do that after the loop with `cns.clear()`